pkg/api/norman/customization/aks/handler.go

```diff
@@ -242,3 +260,33 @@ func handleErr(writer http.ResponseWriter, errorCode int, originalErr error) {
 	writer.WriteHeader(errorCode)
 	writer.Write(asJSON)
 }
+
+func getCachedTenantID(secretClient v1.SecretInterface, subscriptionID string, secret *v1.Secret) (string, error) {
```

Comment:
Weren't we going to reuse the aks-operator one ?

Reply:
Was referencing this: https://github.com/rancher/aks-operator/pull/44#discussion_r675221614
You could just move the update logic out of there. You'd actually be able to sort through errors easier that way, and only ignore the update errors and not the query errors.

Reply:
good point, done